pkg/operator/secretannotator/aws/reconciler.go

```diff
@@ -178,8 +220,55 @@ func (r *ReconcileCloudCredSecret) updateSecretAnnotations(secret *corev1.Secret
 		secretAnnotations = map[string]string{}
 	}
 
-	secretAnnotations[secretconstants.AnnotationKey] = value
+	secretAnnotations[constants.AnnotationKey] = value
 	secret.SetAnnotations(secretAnnotations)
 
 	return r.Update(context.Background(), secret)
 }
+
+// earlyAnnotationIfModeForced will read the CCO config, and go ahead and annotate the secret if
+// there is a forcedCredentialsMode set. It will return the fetched mode (empty string if no mode set),
+// whether the operator is in manual/disabled mode, and any errors encountered.
+func (r *ReconcileCloudCredSecret) earlyAnnotationIfModeForced(secretKey types.NamespacedName) (modeString string, disabled bool, err error) {
+	var mode operatorv1.CloudCredentialsMode
+	mode, err = utils.GetOperatorMode(r.Client, r.Logger)
+	if err != nil {
+		return
+	}
+
+	disabled, err = utils.IsOperatorDisabled(r.Client, r.Logger)
```

Comment:
Should it be an error if the mode was explicitly set to something other than "manual" and there is a configmap setting the operator to disabled?

Reply:
I've been treating the presence of an entry in the config CR as taking precedence over any legacy ConfigMap value.
WDYT about printing a warning if that is detected (explaining that the value in the config CR is being used)?

Reply:
My thinking here is that the ConfigMap is something that the user had to deliberately create and the Mode is something that the user had to deliberately set. I don't think that one has precedence over the other. If they are both present but request different things, then that is a contradiction that CCO cannot resolve.

Reply:
My concern with that is b/c with the next PR where we start to require the presence of a config CR, on upgrade we will lay down an empty config CR and it would immediately put CCO in a bad state as a result of upgrade.

Reply:
No, an empty config CR and a disabled configmap is fine. The only thing that should be an error is a config CR with the mode set to something other than "manual" and a disabled configmap.

Reply:
I see where that last statement is a bit confusing. By "set to something other than manual", I mean a config CR that has a mode set (ie, not empty) and the mode is something other than "manual".

Reply:
Config CR Mode | Disabled ConfigMap | Behavior
---- | ---- | ----
"" | no | permissions checks
"" | yes | manual
"manual" | no | manual
"manual" | yes | manual
"passthrough" | no | passthrough
"passthrough" | yes | error